### PR TITLE
🧹 doc: update connector documentation about CAT and airbyte-ci (new build flow)

### LINKED ISSUE #30767: Modernize Acceptance Test Docs and remove old tools
We should update https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/connector-acceptance-tests-reference.md to stop talking about docker build and ./acceptance-test-docker.sh... and use airbyte-ci instead.  @gisripa  is building a connector the hard way as part of his onboarding, and this was rather confusing

---

We daggerized CAT and it can now be run inside `airbyte-ci`
It also can be run directly from its poetry package with: 
`poetry run pytest  -p connector_acceptance_test.plugin --acceptance-test-config=../../connectors/source-postgres`

But I believe we have stale docs and scripts that can make running CAT confusing.
E.G: we still have `acceptance-test-docker.sh` scripts.

Definition of done:
* The one or two good ways to run CAT are documented
* We delete the CAT dockerfile
* The `acceptance-test-docker.sh` scripts are gone

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes #30767
Our connectors README were stale because:
* Connectors are not built via docker build or (gradle for python connectors) anymore
* CAT is not run via docker run but wrapped into `airbyte-ci`, which makes its Dockerfile and `acceptance-test-docker.sh` script legacy

## How
* Remove all reference to Gradle from the python connector README
* Update README to reflect the up-to-date way of building/testing/publishing connectors
* Update connector generator templates accordingly
* Remove all `acceptance-test-docker.sh` scripts
* Remove the CAT Dockerfile: it's built and run via airbyte-ci
* Remove all "migrations" script we had in connector-acceptance-test/tool: they're using CAT run via docker and any new migration flow will likely by done by introducing a new `airbyte-ci` command.

### CODE REVIEW

**Comment by alafanechere** [comment]
Current dependencies on/for this PR:
* master
  * **PR #32199** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/32199" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/32199?utm_source=stack-comment).

File: airbyte-integrations/connectors/source-scaffold-java-jdbc/metadata.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ data:
       enabled: false
   connectorSubtype: database
   connectorType: source
-  definitionId: FAKE-UUID-0000-0000-000000000000
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-scaffold-java-jdbc/metadata.yaml:12]: please revert

Comment (stephane-airbyte) [reply airbyte-integrations/connectors/source-scaffold-java-jdbc/metadata.yaml:12]: or if needs to be modified, please do that as part of another PR

**Review by stephane-airbyte — CHANGES_REQUESTED** [review]
Looks like all the README.md modifications and the acceptance-test-docker.sh removal belong together. There's one build.gradle that needs to be reverted, and I'd like to see the metadata.yaml modified in a different PR, so that this big PR (in term of # of files modfified) doesn't pollute the history of files that will often be looked at

File: airbyte-integrations/connectors/source-scaffold-java-jdbc/build.gradle [hunk 2]
```diff
@@ -6,7 +6,7 @@ plugins {
 airbyteJavaConnector {
     cdkVersionRequired = '0.2.0'
     features = ['db-sources']
-    useLocalCdk = false
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-integrations/connectors/source-scaffold-java-jdbc/build.gradle:9]: This is a generated change. [The java-jdbc connector template sets this to true](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/source-java-jdbc/build.gradle#L9). Let me know if this is wrong, but it's not a change I made in this PR.

**Comment by alafanechere** [comment]
> There's one build.gradle that needs to be reverted, and I'd like to see the metadata.yaml modified in a different PR, so that this big PR (in term of # of files modfified) doesn't pollute the history of files that will often be looked at

@stephane-airbyte these are "generated changes". Scaffolds connectors are used to test our generator output.
These change were made following my java connector template change:
` CI=true ./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` 
If `useLocalCdk` is set to `true` it's because the [template](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/source-java-jdbc/build.gradle#L9) sets it to true

**Comment by stephane-airbyte** [comment]
> > There's one build.gradle that needs to be reverted, and I'd like to see the metadata.yaml modified in a different PR, so that this big PR (in term of # of files modfified) doesn't pollute the history of files that will often be looked at
> 
> @stephane-airbyte these are "generated changes". Scaffolds connectors are used to test our generator output. These change were made following my java connector template change: ` CI=true ./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` If `useLocalCdk` is set to `true` it's because the [template](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/source-java-jdbc/build.gradle#L9) sets it to true

in that case, the main point of my request still stands. Those need to be part of their own PR, and documented as automated changes in the PR description.

**Pull request closed without merging**